source/fundamentals/crud/read-operations/count.txt

```diff
@@ -0,0 +1,246 @@
+   improve the performance of this method by using a hint to take advantage of the built-in index on
+   the ``_id`` field. Use this technique only when calling ``CountDocuments()``
+   with an empty query parameter.
+
+   .. code-block:: csharp
+      :emphasize-lines: 1, 2
+
+      var filter = Builders<Student>.Filter.Empty;
+      CountOptions opts = new CountOptions(){Hint = "_id_"};
+      var count = collection.CountDocuments(filter, opts);
+
+.. _csharp-estimated-count:
+
+Estimated Count
+---------------
+
+To estimate the total number of documents in your collection, use the
+``EstimatedDocumentCount()`` method. 
+
+.. note:: 
+
+    The ``EstimatedDocumentCount()`` method is more efficient than the
+    ``CountDocuments()`` method because it uses the collection's
+    metadata rather than scanning the entire collection. 
+
+Modify Behavior
+~~~~~~~~~~~~~~~
+
+You can modify the behavior of ``EstimatedDocumentCount()`` by passing a
+``EstimatedDocumentCountOptions`` type as a parameter. If you don't
+specify any options, the driver uses default values.
+
+You can set the following properties in a ``EstimatedDocumentCountOptions`` object:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``MaxTime``
+     - | The maximum amount of time that the query can run on the server.
+       | Default: ``nil``
+
+Example
+```````
+
+The following example estimates the number of documents in the
+``students`` collection:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var count = _myColl.EstimatedDocumentCount();
+      Console.WriteLine("Estimated number of documents in the students collection: " + count);
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Estimated number of documents in the students collection: 6
+
+.. _csharp-count-aggregation:
+
+Aggregation
+-----------
+
+You can use the ``Count()`` builder method to count the number
+of documents in an aggregation pipeline.
+
+Example
+~~~~~~~
+
+The following example performs the following actions:
+
+- Specifies a match stage to find documents with a ``FinalGrade`` value
+  greater than ``80``
+- Counts the number of documents that match the criteria
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var filter = Builders<Student>
+        .Filter.Gt(s => s.FinalGrade, 80);
+      var result = _myColl.Aggregate().Match(filter).Count();
+      Console.WriteLine("Number of documents with a final grade more than 80: " + result.First().Count);
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Number of documents with a final grade more than 80: 2
+      
+
+Additional Information
+----------------------
+
+To learn more about the operations mentioned, see the following
+guides:
+
+- :ref:`csharp-specify-query`
+- :ref:`csharp-bson`
+- :ref:`csharp-guids`
+- :ref:`csharp-builders` 
+- :ref:`csharp-poco`
```

Comment:
Suggestion: add a link to the [$count documentation](https://www.mongodb.com/docs/manual/reference/operator/aggregation/count/) in the Server manual